password.go

```diff
@@ -17,9 +20,49 @@ var PasswordQuestionTemplate = `
 {{- color "green+hb"}}? {{color "reset"}}
 {{- color "default+hb"}}{{ .Message }} {{color "reset"}}`
 
+type filteredReadlinePasswordOutput struct {
+	len int
+}
+
+// Write will be called by readline to echo the masked characters to the console.  It assumes
+// full control of the output with prompting, but since survey is printing the prompt readline
+// does not play nice here.  Write is called with clear-line statements which need to be
+// ignored then it is called with the full buffered output. So if you type "hi" this is the
+// sequence that will be sent to Write
+// 0: "\033[J\033[2K\r"
+// 1: " \b"
+// 2: "\033[J\033[2K\r"
+// 3: "*" (this is the masked "h" from "hi")
+// 4: "\033[J\033[2K\r"
+// 5: "**" (this is the masked "hi" from "hi")
+func (o *filteredReadlinePasswordOutput) Write(p []byte) (n int, err error) {
```

Comment:
If help text is not directly associated with the prompt I am not sure how it could be integrated.  It is necessarily contextual to the question asked by the prompt.  FWIW, I have been doing this with some fugly in-house code (that I am trying to replace with survey) and have had very positive user feedback.

Yeah, I would do it for select and multi-select also, we would just capture the '?' key-press like we capture the arrow keys and space key (which are also inputs the user cannot see).  I have played with the implementation a bit and I think it will be okay (and intuitive).  Without setting `Help` it will be identical to the current behavior.